spec/models/green_flag/feature_spec.rb

```diff
@@ -1,135 +1,187 @@
+      expect{ subject }.to change { GreenFlag::Feature.count }.by(1)
+    end
+    it 'when feature does exists, does not create a feature' do
+      FactoryGirl.create(:green_flag_feature, code: 'foo')
+      expect{ subject }.to_not change { GreenFlag::Feature.count }
+    end
+  end
+
+  describe '#decide_if_enabled_for_visitor' do
+    let(:feature)          { GreenFlag::Feature.create(code: 'big_font') }
+    let(:site_visitor_id)  { 1 }
+
+    let(:non_applying_rule) { double(id: 1, applies_to?: false, decision?: true)  }
+    let(:true_rule)         { double(id: 2, applies_to?: true,  decision?: true)  }
+    let(:false_rule)        { double(id: 3, applies_to?: true,  decision?: false) }
+
+    context 'there are no rules' do
+      it 'returns a feature decision with the correct attributes' do
+        feature_decision = feature.decide_if_enabled_for_visitor(site_visitor_id)
+
+        expect(feature_decision.rule_id).to be_nil
+        expect(feature_decision.enabled).to be_nil
       end
-      it 'when feature does exists, does not create a feature' do
-        FactoryGirl.create(:green_flag_feature, code: 'foo')
-        expect{ subject }.to_not change { GreenFlag::Feature.count }
+    end
+
+    context 'the first applying rule is true' do
+      it 'returns a feature decision with the correct attributes' do
+        feature.stub(rules: [non_applying_rule, true_rule])
+
+        feature_decision = feature.decide_if_enabled_for_visitor(site_visitor_id)
+
+        expect(feature_decision.rule_id).to eq true_rule.id
+        expect(feature_decision.enabled).to eq true
       end
     end
 
-    describe '#decide_if_enabled_for_visitor' do
-      let(:feature)          { GreenFlag::Feature.create(code: 'big_font') }
-      let(:site_visitor_id)  { 1 }
+    context 'the first applying rule is false' do
+      it 'returns a feature decision with the correct attributes' do
+        feature.stub(rules: [false_rule, true_rule])
 
-      let(:non_applying_rule) { double(id: 1, applies_to?: false, decision?: true)  }
-      let(:true_rule)         { double(id: 2, applies_to?: true,  decision?: true)  }
-      let(:false_rule)        { double(id: 3, applies_to?: true,  decision?: false) }
+        feature_decision = feature.decide_if_enabled_for_visitor(site_visitor_id)
 
-      context 'there are no rules' do
-        it 'returns a feature decision with the correct attributes' do
-          feature_decision = feature.decide_if_enabled_for_visitor(site_visitor_id)
+        expect(feature_decision.rule_id).to eq false_rule.id
+        expect(feature_decision.enabled).to eq false
+      end
+    end
+  end
 
-          expect(feature_decision.rule_id).to be_nil
-          expect(feature_decision.enabled).to be_nil
-        end
+  describe '#forget_non_manual_decisions!' do
+    let(:feature) { GreenFlag::Feature.create(code: 'big_font') }
+    let(:enabled) { true }
+
+    context 'when there are no decisions' do
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.